staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -599,6 +600,13 @@ func (c *Cacher) GetToList(ctx context.Context, key string, resourceVersion stri
 		return c.storage.GetToList(ctx, key, resourceVersion, pred, listObj)
 	}
 
+	if resourceVersion == "" {
+		resourceVersion, err := c.storage.GetLastRevision(ctx, key)
+		if err != nil {
+			return c.storage.GetToList(ctx, key, resourceVersion, pred, listObj)
+		}
+	}
+
```

Comment:
I think I don't understand what you're really trying to do. Can you please write down a proposal first?

Also, looking into PR title, https://github.com/kubernetes/enhancements/pull/1404 may be related.